src/analysis/normed_space/weak_dual.lean

```diff
@@ -138,3 +142,158 @@ end
 end normed_space.dual
 
 end weak_star_topology_for_duals_of_normed_spaces
+
+section polar_sets_in_weak_dual
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+variables {E : Type*} [normed_group E] [normed_space 𝕜 E]
+
+open metric set
+
+/-- Given a subset `s` in a normed space `E` (over a field `𝕜`), the polar
+`polar s` is the subset of `weak_dual 𝕜 E` consisting of those functionals which
+evaluate to something of norm at most one at all points `z ∈ s`. -/
+def polar (s : set E) : set (weak_dual 𝕜 E) :=
```

Comment:
Since Lean can not guess the field, I'd make it explicit in the definition of polar, i.e., `polar 𝕜 s`. This should save you a lot of `@` below.

Reply:
Ok, good idea! I did this with `polar` and `eval_bounded_of_nbhd_zero`.